src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/AggregationManager.cs

```diff
@@ -0,0 +1,274 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Diagnostics.Metrics
+{
+    internal class AggregationManager
+    {
+        // these fields are modified after construction and accessed on multiple threads, use lock(this) to ensure the data
+        // is synchronized
+        private List<Predicate<Instrument>> _instrumentConfigFuncs = new();
+        private TimeSpan _collectionPeriod;
+
+        private ConcurrentDictionary<Instrument, InstrumentState> _instrumentStates = new ConcurrentDictionary<Instrument, InstrumentState>();
```

Comment:
Nit:
```suggestion
        private ConcurrentDictionary<Instrument, InstrumentState> _instrumentStates = new();
```
Other fields also use the short form.